src/test/java/hudson/plugins/git/UserRemoteConfigRefSpecTest.java

```diff
@@ -0,0 +1,91 @@
+package hudson.plugins.git;
+
+import hudson.util.FormValidation;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class UserRemoteConfigRefSpecTest {
+
+    @Issue("JENKINS-57660")
+    @Test
+    public void testdoCheckRefspecSuccessWithoutWildcards(){
+        String url = "git://git.example.com/repository-that-does-not-exist";
+        String name = "origin";
+        List<String> refSpec = new ArrayList<>();
+        refSpec.add("+refs/heads/master:refs/remotes/origin/master");
+        refSpec.add("+refs/heads/JENKINS-57660:refs/remotes/origin/JENKINS-57660");
+        refSpec.add("master:refs/remotes/origin/mymaster");
+        refSpec.add("master:refs/remotes/origin/mymaster topic:refs/remotes/origin/topic");
```

Comment:
I missed previously that there is no test which asserts the path for '$' in the refspec.  Could you either add a test for that case or extend one of the existing tests for it?

Reply:
I would love to have that one additional case covered in a test, but will also merge this without that test if you prefer.  Thanks again for the pull request!

Reply:
No, I am adding this case asap and pushing the changes and after that you can merge it at your convenience. Thanks!